frontend/lib/resolution/Resolver.cpp

```diff
@@ -3302,6 +3300,50 @@ void Resolver::exit(const Try* node) {
   exitScope(node);
 }
 
+bool Resolver::enter(const Catch* node) {
+  enterScope(node);
+  if (scopeResolveOnly) {
+    return true;
+  }
+
+  if (auto errVar = node->error()) {
+    CHPL_ASSERT(errVar->initExpression() == nullptr &&
+      "catch variable should not have an init expression");
+
+    const AstNode* typeExpr = errVar->typeExpression();
+    if (typeExpr == nullptr) {
+      // default to Error placeholder
+      const ClassType* errorType = CompositeType::getErrorType(context);
+      auto qt = QualifiedType(QualifiedType::VAR, errorType);
+      ResolvedExpression& re = byPostorder.byAstExpanding(errVar);
+      re.setType(qt);
+    } else {
+      errVar->traverse(*this);
+    }
+
+    ResolvedExpression& re = byPostorder.byAst(errVar);
+
+    if(auto ct = re.type().type()->toClassType()) {
```

Comment:
What are we doing if the type _isn't_ the `Error` type (or a subclass of `Error`?). Maybe check production to see what happens if you try to type `e` as `int` or something? We should probably handle that case here. You can just do a `context->error` if you don't want to do a named error.